bark/world/evaluation/rss/rss_interface.cpp

```diff
@@ -0,0 +1,477 @@
+          relevant_distance)
+        relevent_agents.push_back(other_agent.second);
+    }
+  }
+
+  // +1 is a work around because RSS defines the world is only valid after
+  // world_time >=1
+  ::ad::rss::map::RssSceneCreation scene_creation(ego_rss_state.timestamp + 1,
+                                                  ego_dynamics);
+
+  // It is not relevent, but needed by appendScenes.
+  ::ad::map::landmark::LandmarkIdSet green_traffic_lights;
+
+  for (const auto& other : relevent_agents) {
+    models::dynamic::State other_state = other->GetCurrentState();
+
+    Polygon other_shape = other->GetShape();
+    auto const other_match_object =
+        GenerateMatchObject(other_state, other_shape);
+
+    ::ad::rss::world::RssDynamics other_dynamics =
+        GenerateAgentDynamicsParameters(other->GetAgentId());
+    auto other_av =
+        CaculateAgentAngularVelocity(other->GetExecutionTrajectory());
+
+    ::ad::rss::map::RssObjectData other_data = GenerateObjectData(
+        ::ad::rss::world::ObjectId(other->GetAgentId()),
+        ::ad::rss::world::ObjectType::OtherVehicle, other_match_object,
+        other_state(VEL_POSITION), other_av,
+        ::ad::physics::Angle(other_state(THETA_POSITION)), other_dynamics);
+
+    // Find all possible scenes between the ego agent and a relevent agent
+    scene_creation.appendScenes(ego_data, ego_route, other_data,
+                                ::ad::rss::map::RssSceneCreation::
+                                    RestrictSpeedLimitMode::ExactSpeedLimit,
+                                green_traffic_lights,
+                                ::ad::rss::map::RssMode::Structured);
+  }
+  rss_world = scene_creation.getWorldModel();
+
+  // It is valid only after world timestep 1
+  return withinValidInputRange(rss_world);
+}
+
+bool RssInterface::RssCheck(
+    const ::ad::rss::world::WorldModel& world_model,
+    ::ad::rss::state::RssStateSnapshot& rss_state_snapshot) {
+  ::ad::rss::core::RssCheck rss_check;
+  // Describes the relative relation between two objects in possible situations
+  ::ad::rss::situation::SituationSnapshot situation_snapshot;
+
+  // Contains longitudinal and lateral response of the ego object, a list of id
+  // of the dangerous objects
+  ::ad::rss::state::ProperResponse proper_response;
+
+  // rss_state_snapshot: individual situation responses calculated from
+  // SituationSnapshot
+  bool result = rss_check.calculateProperResponse(
+      world_model, situation_snapshot, rss_state_snapshot, proper_response);
+
+  if (!result) {
+    LOG(ERROR) << "Failed to perform RSS check" << std::endl;
+  }
+
+  return result;
+}
+
+bool RssInterface::ExtractSafetyEvaluation(
+    const ::ad::rss::state::RssStateSnapshot& snapshot) {
+  bool safety_response = true;
+  for (auto const state : snapshot.individualResponses) {
+    safety_response = safety_response && !::ad::rss::state::isDangerous(state);
```

Comment:
safety_response means then that we need to enforce safety measures?

Reply:
No, safety_response is defined by as if the ego is at a overally safe situation. I guess I don't have a good sense for naming😅.
Is it better to check it to `is_safe`?

Reply:
Okay, just have to remember that, even though a documentation of this would be nice; )